src/Microsoft.DncEng.SecretManager.Tests/TemporaryFile.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.DncEng.SecretManager.Tests
+{
+    public class TemporaryFile : IDisposable
+    {
+        public string FilePath { get; }
+
+        public TemporaryFile()
+        {
+            FilePath = Path.GetTempFileName();
+        }
+
+        public async Task WriteAllTextAsync(string text)
+        {
+            using var file = new FileStream(FilePath, FileMode.Create, FileAccess.Write);
```

Comment:
If you use "CreateText" it just gives you the StreamWriter and saves you a step. :-)

Reply:
Or just use File.WriteAllTextAsync...

Reply:
Yea, it was initially more complex because I did a FileOptions.DeleteOnClose thing, but now I can clean it up.